### PR TITLE
Catch CalledProcessError in build script.

### PR DESCRIPTION
**Pull request by zhangqiaorjc**
print(e.output) should tell us the build errors for https://github.com/google/jax/issues/6637

### CODE REVIEW

File: build/build.py [hunk 1]
```diff
@@ -48,7 +48,10 @@ def is_windows():
 
 
 def shell(cmd):
-  output = subprocess.check_output(cmd)
+  try:
+    output = subprocess.check_output(cmd)
+  except subprocess.CalledProcessError as e:
+    print(e.output)
```
Comment (hawkinsp) [inline build/build.py:54]: I think we should reraise the exception here?

File: build/build.py [hunk 2]
```diff
@@ -206,7 +209,8 @@ def get_bazel_path(bazel_path_flag):
 def check_bazel_version(bazel_path):
   try:
     version_output = shell([bazel_path, "--bazelrc=/dev/null", "version"])
-  except subprocess.CalledProcessError:
+  except subprocess.CalledProcessError as e:
```
Comment (hawkinsp) [inline build/build.py:212]: Isn't this one redudant if you rethrow the exception above?

File: build/build.py (hunk 1 above)
Comment (zhangqiaorjc) [reply build/build.py:54]: Done

File: build/build.py (hunk 2 above)
Comment (zhangqiaorjc) [reply build/build.py]: Reverted

File: build/build.py (hunk 1 above)
Comment (hawkinsp) [reply build/build.py:54]: I think to reraise you can just write `raise` with no `e`.

File: build/build.py [hunk 3]
```diff
@@ -48,7 +48,11 @@ def is_windows():
 
 
 def shell(cmd):
-  output = subprocess.check_output(cmd)
+  try:
+    output = subprocess.check_output(cmd)
+  except subprocess.CalledProcessError as e:
+    print(e.output)
```
Comment (hawkinsp) [inline build/build.py:54]: Shouldn't this be `output` not `e.output`?

Verify this works on linux?

Comment (hawkinsp) [reply build/build.py:54]: Actually I guess this is fine as is.

File: build/build.py (hunk 1 above)
Comment (zhangqiaorjc) [reply build/build.py:54]: done

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**